src/toast/pipeline_tools/binning.py

```diff
@@ -0,0 +1,290 @@
+    try:
+        parser.add_argument(
+            "--hits",
+            required=False,
+            action="store_true",
+            help="Write hit maps [default]",
+            dest="write_hits",
+        )
+        parser.add_argument(
+            "--no-hits",
+            required=False,
+            action="store_false",
+            help="Do not write hit maps",
+            dest="write_hits",
+        )
+        parser.set_defaults(write_hits=True)
+    except argparse.ArgumentError:
+        pass
+
+    try:
+        parser.add_argument(
+            "--wcov",
+            required=False,
+            action="store_true",
+            help="Write white noise covariance [default]",
+            dest="write_wcov",
+        )
+        parser.add_argument(
+            "--no-wcov",
+            required=False,
+            action="store_false",
+            help="Do not write white noise covariance",
+            dest="write_wcov",
+        )
+        parser.set_defaults(write_wcov=True)
+    except argparse.ArgumentError:
+        pass
+
+    try:
+        parser.add_argument(
+            "--wcov-inv",
+            required=False,
+            action="store_true",
+            help="Write inverse white noise covariance [default]",
+            dest="write_wcov_inv",
+        )
+        parser.add_argument(
+            "--no-wcov-inv",
+            required=False,
+            action="store_false",
+            help="Do not write inverse white noise covariance",
+            dest="write_wcov_inv",
+        )
+        parser.set_defaults(write_wcov_inv=True)
+    except argparse.ArgumentError:
+        pass
+
+    try:
+        parser.add_argument(
+            "--zip",
+            required=False,
+            action="store_true",
+            help="Compress the map outputs",
+            dest="zip_maps",
+        )
+        parser.add_argument(
+            "--no-zip",
+            required=False,
+            action="store_false",
+            help="Do not compress the map outputs",
+            dest="zip_maps",
+        )
+        parser.set_defaults(zip_maps=True)
+    except argparse.ArgumentError:
+        pass
+
+    # `nside` may already be added
+    try:
+        parser.add_argument(
+            "--nside", required=False, default=512, type=np.int, help="Healpix NSIDE"
+        )
+    except argparse.ArgumentError:
+        pass
+    return
+
+
+def init_binner(args, comm, data, detweights, subnpix=None, localsm=None, verbose=True):
+    """construct distributed maps to store the covariance,
+    noise weighted map, and hits
+
+    Args:
+        detweights (dict) :  A dictionary of the dimensional, inverse variance
+             detector weights that form the diagonal of N^-1.
+    Returns:
+        npp (DistPixels) :  The white noise covariance matrices needed to
+            bin signal onto the distributed map `zmap`.
+        zmap (DistPixels) :  An empty distributed map compatible with `npp`
```

Comment:
what if we name `zmap` -> `dist_map` and `npp` -> `white_noise_cov_matrix`

Reply:
Looks a little clunky but definitely more descriptive!